utils/v2tofhirr4/mapping_defs.bal

```diff
@@ -90,23 +90,20 @@ public type SegmentToFhir record {
     OrcToImmunization orcToImmunization?;
 };
 
-# Context class to hold v2tofhir mapper related context information.
-public class V2ToFhirMapperContext {
-    private SegmentToFhir mapperImpl = {
-        pv1ToPatient: pv1ToPatient,
-        pv1ToEncounter: pv1ToEncounter,
-        nk1ToPatient: nk1ToPatient,
-        pd1ToPatient: pd1ToPatient,
-        pidToPatient: pidToPatient,
-        dg1ToCondition: dg1ToCondition,
-        obxToObservation: obxToObservation,
-        obrToDiagnosticReport: obrToDiagnosticReport,
-        al1ToAllerygyIntolerance: al1ToAllerygyIntolerance,
-        evnToProvenance: evnToProvenance,
-        mshToMessageHeader: mshToMessageHeader,
-        pv2ToEncounter: pv2ToEncounter,
-        orcToImmunization: orcToImmunization
-    };
-
-    function getDefaultImpl() returns SegmentToFhir => self.mapperImpl;
-}
+// Record initialized with the default mapping functions.
+isolated V2SegmentToFhirMapper defaultMapper = {
```

Comment:
Does this have to be mutable? Can't it be final and readonly? 

(Haven't checked each type).

Reply:
Can get rid of clones if so too.

Reply:
This is assuming V2SegmentToFhirMapper has all function pointers. You'll have to fix the rest descriptor too (instead of the default anydata).

Reply:
This is a mutable instance where users can pass the overriden functions that will update the current function impls.

Reply:
Does it update this defaultMapper value? Is that a future change? Asking because there seem to be clones wherever it is used atm.

Reply:
Yes, it updates the default mapper.

Reply:
How is it that mapping created though? Do you directly update `defaultMapper`? Currently seems to be using clones always?